apps/emqx_message_transformation/src/emqx_message_transformation_http_api.erl

```diff
@@ -0,0 +1,656 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-module(emqx_message_transformation_http_api).
+
+-behaviour(minirest_api).
+
+-include_lib("typerefl/include/types.hrl").
+-include_lib("hocon/include/hoconsc.hrl").
+-include_lib("emqx/include/logger.hrl").
+-include_lib("emqx_utils/include/emqx_utils_api.hrl").
+
+%% `minirest' and `minirest_trails' API
+-export([
+    namespace/0,
+    api_spec/0,
+    fields/1,
+    paths/0,
+    schema/1
+]).
+
+%% `minirest' handlers
+-export([
+    '/message_transformations'/2,
+    '/message_transformations/reorder'/2,
+    '/message_transformations/transformation/:name'/2,
+    '/message_transformations/transformation/:name/metrics'/2,
+    '/message_transformations/transformation/:name/metrics/reset'/2,
+    '/message_transformations/transformation/:name/enable/:enable'/2
```

Comment:
Would be nice to "test" transformation(s) with a given message, but that probably should be dashboard's team requirement

Reply:
Yes, that's a follow up PR to implement the test endpoint.